packages/data/README.md

```diff
@@ -793,9 +793,25 @@ any price in the state for that currency is retrieved. If the currency prop
 doesn't change and other props are passed in that do change, the price will
 not change because the dependency is just the currency.
 
+When data is only used in an event callback, the data should not be retrieved
+on render, so it may be useful to get the selectors function instead.
```

Comment:
I think we should say something like: don't use this API while calling the selectors directly on the render function,   your component won't rerender when changes happen.

Reply:
This actually makes me wonder: why should we retrieve the selectors during render at all if we could also retrieve them in the callback itself? In other words, in the callbacks, we could just use `const { selector } = select( key )`?

Reply:
The global  `select/dispatch/subscribe` function should be avoided as much as we can. These don't rely on the RegistryProvider. For me ideally, these should become WP-only APIs added as inline styles and not something the package exports by default.

Reply:
That would have the additional benefit of not having to pass the function as a dependency of callbacks. I'll explore this quickly in a separate PR.

Reply:
I think you misread my comment, we shouldn't do it :)

Reply:
Oh, Github didn't show me your comment at the time so I didn't see it :)